### PR TITLE
AM-609_Fix_Sonar_Issues

### PR DESCRIPTION
**Pull request by nitinsahu1**
JIRA link (if applicable)
https://tools.hmcts.net/jira/browse/AM-609

Change description
Code smell and security issues as per Sonar

Does this PR introduce a breaking change? (check one with "x")

[ ] Yes
[ x] No

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/ccd/document/am/Application.java [hunk 1]
```diff
@@ -21,7 +21,7 @@
 public class Application {
 
     public static void main(final String[] args) {
-        SpringApplication.run(Application.class, args);
+        SpringApplication.run(Application.class);
```
Comment (dharmendra8186) [inline src/main/java/uk/gov/hmcts/reform/ccd/document/am/Application.java]: If We need to pass java command line parameter in future.How will we pass to it as you remove args parameter?

File: src/main/java/uk/gov/hmcts/reform/ccd/document/am/controller/WelcomeController.java [hunk 2]
```diff
@@ -43,7 +38,7 @@
         return ok("Welcome to CCD Case Document AM Controller");
     }
 
-    @RequestMapping("/swagger")
+    @GetMapping(value = "/swagger")
     public String index() {
```
Comment (dharmendra8186) [inline src/main/java/uk/gov/hmcts/reform/ccd/document/am/controller/WelcomeController.java]: We really need this end point.As, Swagger itself provides default end points.

File: src/main/java/uk/gov/hmcts/reform/ccd/document/am/service/impl/CaseDataStoreServiceImpl.java [hunk 3]
```diff
@@ -69,23 +71,23 @@ public CaseDataStoreServiceImpl(RestTemplate restTemplate, SecurityUtils securit
...
+                    throw new ForbiddenException(CASE_ERROR_MESSAGE + caseId);
                 }
                 return Optional.of(caseDocumentMetadata);
             }
         } catch (HttpClientErrorException exception) {
             if (HttpStatus.NOT_FOUND.equals(exception.getStatusCode())) {
-                LOG.error("Could't find document for case  : " + caseId + ", response code from CCD : " + HttpStatus.NOT_FOUND);
-                throw new ForbiddenException("Could't find document for case  : " + caseId);
+                LOG.error(ERROR_MESSAGE, caseId, HttpStatus.NOT_FOUND);
+                throw new ForbiddenException(CASE_ERROR_MESSAGE + caseId);
```
Comment (dharmendra8186) [inline src/main/java/uk/gov/hmcts/reform/ccd/document/am/service/impl/CaseDataStoreServiceImpl.java]: We are checking 404(Not Found) and throwing 403 (Forbidden) Exception.It looks very weird.

File: src/main/java/uk/gov/hmcts/reform/ccd/document/am/Application.java (hunk 1 above)
Comment (nitinsahu1) [reply src/main/java/uk/gov/hmcts/reform/ccd/document/am/Application.java]: Sonar does not allow unsafe command line parms and we should make sure that command line arguments are used safely here. e.g. we should sanitise it manually and either fix it if it’s a real vulnerability or resolve it as reviewed if it’s safe.

File: src/main/java/uk/gov/hmcts/reform/ccd/document/am/controller/WelcomeController.java (hunk 2 above)
Comment (nitinsahu1) [reply src/main/java/uk/gov/hmcts/reform/ccd/document/am/controller/WelcomeController.java]: Above comments can be incorporated as part of clean up task.

File: src/main/java/uk/gov/hmcts/reform/ccd/document/am/service/impl/CaseDataStoreServiceImpl.java (hunk 3 above)
Comment (nitinsahu1) [reply src/main/java/uk/gov/hmcts/reform/ccd/document/am/service/impl/CaseDataStoreServiceImpl.java]: Valid comment and can be incorporated in functional test task.

**Pull request merged**